### PR TITLE
Remove core_extensions from `mail` when running Ruby > 2.0

### PR DESCRIPTION
**Pull request by metcalf**
This PR removes all of the remaining core_extensions from mail, excluding backports from newer versions of Ruby. Each commit in this series removes a specific extension. I thought this would be easier to review as a series of commits than as a series of totally separate PRs but happy to switch.

### CODE REVIEW

File: lib/mail/core_extensions/string/access.rb [hunk 1]
```diff
@@ -7,48 +7,6 @@
 # itself does not depend on ActiveSupport to avoid versioning conflicts
 
 class String
-  unless '1.9'.respond_to?(:force_encoding)
```
Comment (metcalf) [inline lib/mail/core_extensions/string/access.rb]: These were unused as well. In #1094 I thought `iconv` returned a String but I confirmed that it returns an array so `first` is never called on String.

https://github.com/mikel/mail/blob/2b76d0a349895a84f02f1f86f762a4906bd133e8/lib/mail/encodings.rb#L153

File: lib/mail/multibyte/chars.rb [hunk 2]
```diff
@@ -103,23 +103,23 @@ def self.wants?(string)
         # Returns a new Chars object containing the _other_ object concatenated to the string.
         #
         # Example:
-        #   ('Café'.mb_chars + ' périferôl').to_s # => "Café périferôl"
```
Comment (metcalf) [inline lib/mail/multibyte/chars.rb]: Note that all the changes in this file are just rewriting comments.

Comment (jeremy) [inline lib/mail/multibyte/chars.rb]: Thanks for the heads up 😊

File: spec/mail/utilities_spec.rb [hunk 3]
```diff
@@ -179,13 +179,13 @@
...
       result = 'This is a string'
       expect(unbracket(test)).to eq result
     end
 
     it "should work without parens using ActiveSupport mb_chars" do
-      test = 'This is a string'.mb_chars
+      test = Mail::Multibyte.mb_chars('This is a string')
       result = 'This is a string'
       expect(unbracket(test)).to eq(result)
     end
```
Comment (jeremy) [inline spec/mail/utilities_spec.rb]: These aren't really testing ActiveSupport::Multibyte chars, huh?

File: lib/mail.rb [hunk 4]
```diff
@@ -35,8 +35,6 @@ module Mail # :doc:
   if defined?(ActiveSupport)
     require 'active_support/inflector'
   else
-    require 'mail/core_extensions/string/access'
-    require 'mail/core_extensions/string/multibyte'
```
Comment (jeremy) [inline lib/mail.rb]: 🤘

File: lib/mail/core_extensions/smtp.rb [hunk 5]
```diff
@@ -1,25 +1,28 @@
...
-        begin
-          original_tlsconnect(s).tap { verified = true }
-        ensure
-          unless verified
-            s.close rescue nil
+# This is a backport of r30294 from ruby trunk because of a bug in net/smtp.
+# http://svn.ruby-lang.org/cgi-bin/viewvc.cgi?view=rev&amp;revision=30294
+#
+# Fixed in Ruby 1.9.3 - tlsconnect also does not exist in some early versions of ruby
+if RUBY_VERSION < '1.9.3'
```
Comment (jeremy) [inline lib/mail/core_extensions/smtp.rb]: Note that we already do this version check in https://github.com/mikel/mail/blob/master/lib/mail.rb#L31. It's clearer to do it within this file, though! Let's drop the conditional in mail.rb.

Comment (metcalf) [inline lib/mail/core_extensions/smtp.rb]: Changed

File: spec/mail/utilities_spec.rb (hunk 3 above)
Comment (metcalf) [inline spec/mail/utilities_spec.rb]: Reworded these to match the current code.

**Comment by jeremy** [comment]
Thanks @metcalf!!

**Comment by metcalf** [comment]
Thank you! Very excited to 💥 these monkeypatches from our codebase.

Would you be willing to cut a release that includes this? If not I can cut one on our internal gem server but it's always nice to keep consistent with public releases.

**Comment by jeremy** [comment]
They'll be included in the [upcoming 2.7.0 minor release](https://github.com/mikel/mail/milestone/5). I'd rather not do these removals in a 2.6.x point release.

**Comment by metcalf** [comment]
Ok, thanks!

**Pull request closed without merging**